gcc/rust/hir/tree/rust-hir-item.h

```diff
@@ -2495,11 +2505,7 @@ class TraitItemConst : public TraitItem
 
   std::unique_ptr<Type> &get_type () { return type; }
 
-  std::unique_ptr<Expr> &get_expr ()
-  {
-    rust_assert (has_expr ());
```

Comment:
@philberty pointing this out, as you may not see this particular change. It makes the life of the dumper easier if any unique_ptr& can be tested before printing it's content. If the getter panics, it means the unique_ptr can't be tested and the dumper can't be generic and has to use the specific test for every classes (`has_expr`, `has_foo`, ...). Only this one caused issue when testing, but maybe there are other

Reply:
Good find i have a feeling i put this assertion in because I was hitting a bug in hir lowering ages ago. I think a getter with this assertion is a little wierd so yeah i approve of this.